core/src/main/java/com/linecorp/armeria/common/logging/ResponseLogLevelMapper.java

```diff
@@ -0,0 +1,120 @@
+
+    /**
+     * Returns the default {@link ResponseLogLevelMapper} which returns the specified
+     * {@code successfulResponseLogLevel} when logging successful responses (e.g., no unhandled exception) and
+     * {@code failureResponseLogLevel} if failure.
+     */
+    static ResponseLogLevelMapper of(LogLevel successfulResponseLogLevel, LogLevel failureResponseLogLevel) {
+        return log -> log.responseCause() == null ? successfulResponseLogLevel : failureResponseLogLevel;
+    }
+
+    /**
+     * Creates a new {@link ResponseLogLevelMapper} which returns the specified {@link LogLevel} if the given
+     * {@link RequestLog#responseStatus()} is equal to the specified {@link HttpStatus}.
+     */
+    static ResponseLogLevelMapper of(HttpStatus status, LogLevel logLevel) {
+        requireNonNull(status, "status");
+        requireNonNull(logLevel, "logLevel");
+        return log -> log.responseStatus() == status ? logLevel : null;
+    }
+
+    /**
+     * Creates a new {@link ResponseLogLevelMapper} which returns the specified {@link LogLevel} if the given
+     * {@link RequestLog#responseStatus()} belongs to the specified {@link HttpStatusClass}.
+     */
+    static ResponseLogLevelMapper of(HttpStatusClass statusClass, LogLevel logLevel) {
+        requireNonNull(statusClass, "statusClass");
+        requireNonNull(logLevel, "logLevel");
+        return log -> log.responseStatus().codeClass() == statusClass ? logLevel : null;
+    }
+
+    /**
+     * Returns a {@link LogLevel} for the given {@link RequestLog}. Returning {@code null} lets the next handler
+     * specified with {@link #orElse(ResponseLogLevelMapper)} map the {@link RequestLog}.
+     */
+    @Nullable
+    @Override
+    LogLevel apply(RequestLog log);
+
+    /**
+     * Returns a composed {@link ResponseLogLevelMapper} which represents a logical OR of this
+     * {@link ResponseLogLevelMapper} and the given {@code other}.
+     * If this {@link ResponseLogLevelMapper#apply(RequestLog)} returns {@code null}, then the other
+     * {@link ResponseLogLevelMapper} will be applied.
+     */
+    default ResponseLogLevelMapper orElse(ResponseLogLevelMapper other) {
+        requireNonNull(other, "other");
+        if (this == other) {
+            return this;
+        }
+        return log -> {
+            final LogLevel logLevel = apply(log);
+            if (logLevel != null) {
+                return logLevel;
+            }
+            return other.apply(log);
+        };
+    }
+
+    /**
+     * Returns a composed function that first applies the before function to its input, and then applies this
+     * function to the result. If evaluation of either function throws an exception, it is relayed to the
+     * caller of the composed function.
+     *
+     * @deprecated Do not use this method.
+     */
+    @Deprecated
+    @Override
+    default <V> Function<V, LogLevel> compose(Function<? super V, ? extends RequestLog> before) {
```

Comment:
Should we apply these to `RequestLogLevelMapper`?

Reply:
Thanks, nice catch (_ _)

Reply:
Additionally, how about throwing an `UnsupportedOperationException`?

Reply:
+1 That sounds good.